packages/admin/admin/src/FinalFormSaveButton.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
+import { useForm, useFormState } from "react-final-form";
+import { FormattedMessage } from "react-intl";
+
+import { SaveButton } from "./common/buttons/save/SaveButton";
+import { messages } from "./messages";
+
+interface FinalFormSaveButtonProps {
+    message?: React.ReactNode;
+}
+
+export const FinalFormSaveButton = ({ message = <FormattedMessage {...messages.save} /> }: FinalFormSaveButtonProps) => {
+    const form = useForm();
+    const { pristine, hasValidationErrors, submitting, hasSubmitErrors } = useFormState();
+
+    return (
+        <SaveButton
+            disabled={pristine || hasValidationErrors || submitting}
+            color="primary"
+            variant="contained"
+            saving={submitting}
+            hasErrors={hasSubmitErrors}
+            onClick={() => {
+                !pristine && !hasValidationErrors && !submitting && form.submit();
```

Comment:
I know this is very opinionated, but I think short-circuiting if's are difficult to read, while adding no real benefit (reducing line count is no benefit). Others agree: https://stackoverflow.com/questions/34704750/one-line-if-vs-in-javascript#:~:text=If%20you%20are%20writing%20an,logical%20conjunction%20of%20two%20values. Would you mind using a good old if? 😁 

Or would you do something like this `(!pristine && !hasValidationErrors && !submitting) && form.submit() || /* else branch */`

Reply:
No it's fine for me. I'll change it 👍🏻